app/policies/wiki_policy.rb

```diff
@@ -0,0 +1,47 @@
+class WikiPolicy < ApplicationPolicy
+
+  def initialize(user, wiki)
+    @user = user
+    @wiki = wiki
+  end
+
+  def destroy?
+    user && (@user.admin? || @wiki.user_id == @user.id)
+  end
+
+
+  class Scope < Scope
+    attr_reader :user, :scope
+
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+
+    def resolve
+      wikis = []
+      if user && user.role == 'admin'
+        wikis = scope.all
+      elsif user && user.role == 'premium'
+        all_wikis = scope.all
+        all_wikis.each do |wiki|
+          if !wiki.private || wiki.user_id == user.id || wiki.collaborators.include?(user_id: user.id)
+            wikis << wiki
+            puts "Collaborator or Admin"
+          end
+        end
+      else
+        all_wikis = scope.all
+        all_wikis.each do |wiki|
+          if !wiki.private || wiki.collaborators.include?(user_id: user.id)
+            if wiki.collaborators.include?(user_id: user.id)
+              puts "\n\nNon Premium Collaborator Found\n\n"
+            end
+            wikis << wiki
+          end
```

Comment:
I would use `map` instead of `each` here.  Also, I would remove the `puts` unless you want that included in the logs.  They won't affect any of the code's functionality.